network/network_windows.go

```diff
@@ -216,8 +226,19 @@ func (nm *networkManager) configureHcnNetwork(nwInfo *NetworkInfo, extIf *extern
 
 	// Set hcn network adaptor name policy
 	// FixMe: Find a better way to check if a nic that is selected is not part of a vSwitch
-	if !strings.HasPrefix(extIf.Name, vEthernetAdapterPrefix) {
-		netAdapterNamePolicy, err := policy.GetHcnNetAdapterPolicy(extIf.Name)
+	// Pass adapter name here if it is not empty, this is cause if we don't tell HNS which adapter to use
+	// it will just pick one randomly, this is a problem for customers that have multiple adapters
+	if nwInfo.AdapterName != "" || !strings.HasPrefix(extIf.Name, vEthernetAdapterPrefix) {
+		var adapterName string
+		if (nwInfo.AdapterName != "") {
```

Comment:
nitpick: remove ( after if, not a golang style.

Reply:
Removed